### PR TITLE
chore: improvements to settings

### PR DESCRIPTION
**Pull request by KaWaite**
This updates the config to better suit the current development style.

Breaking change: 
- Now it is a function that takes in a single projectName param. This allows each team to filter their imports based on the project. Ex @flow, @visualizer, @cms, etc

### CODE REVIEW

File: typescript.mjs [hunk 1]
```diff
@@ -2,6 +2,20 @@ import tseslint from "typescript-eslint";
...
+    "@typescript-eslint/no-explicit-any": "off",
+    "@typescript-eslint/no-unused-vars": [
+      "error",
+      {
+        args: "all",
+        argsIgnorePattern: "^(_|e)",
+        caughtErrors: "all",
+        caughtErrorsIgnorePattern: "^(_|e)",
+        destructuredArrayIgnorePattern: "^(_|e)",
+        varsIgnorePattern: "^(_|e)",
```
Comment (airslice) [inline typescript.mjs:16]:
```suggestion
        argsIgnorePattern: "^_",
        caughtErrors: "all",
        caughtErrorsIgnorePattern: "^_",
        destructuredArrayIgnorePattern: "^_",
        varsIgnorePattern: "^_",
```

File: common.mjs [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import js from "@eslint/js";
 import globals from "globals";
 
-/** @type { import("eslint").Linter.FlatConfig[] } */
```
Comment (keiya01) [inline common.mjs:4]: Why did you changed to use Config instead of FlatConfig?

Comment (KaWaite) [reply common.mjs:4]: Its the reverse. FlatConfig (as far as i know) was a temporary typing and now Config is what needs to be used.

Comment (keiya01) [reply common.mjs:4]: I see. Thank you!

**Review by keiya01 — APPROVED** [review]

**Pull request merged**